MiniProject/Modules/MenuList/Scene/View/MenuListViewController.swift

```diff
@@ -0,0 +1,309 @@
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.backgroundColor = .clear
+        collectionView.alwaysBounceVertical = true
+
+        collectionView.register(MenuItemCell.self, forCellWithReuseIdentifier: MenuItemCell.identifier)
+
+        view.addSubview(collectionView)
+
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: 16),
+            collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+            collectionView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ])
+    }
+    
+    // MARK: - UICollectionViewDataSource
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let selectedItem = filteredMenuItems[indexPath.item]
+        let detailViewController = DetailMenuViewController()
+        detailViewController.configure(with: selectedItem)
+        navigationItem.backButtonTitle = ""
+        navigationController?.pushViewController(detailViewController, animated: true)
+        
+        UIAccessibility.post(notification: .announcement, argument: "Navigating to \(selectedItem.strMeal) details.")
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return filteredMenuItems.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: MenuItemCell.identifier, for: indexPath) as? MenuItemCell else {
+            fatalError("Unable to dequeue MenuItemCell")
+        }
+        let item = filteredMenuItems[indexPath.item]
+        cell.configure(with: item)
+        
+        cell.isAccessibilityElement = true
+        cell.accessibilityLabel = item.strMeal
+        cell.accessibilityHint = "Double-tap to view details about \(item.strMeal)."
+        
+        return cell
+    }
+
+    // MARK: - UICollectionViewDelegateFlowLayout
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let totalSpacing: CGFloat = 16
+        let numberOfItemsPerRow: CGFloat = 2
+        
+        let width = (collectionView.bounds.width - (numberOfItemsPerRow - 1) * totalSpacing) / numberOfItemsPerRow
+        let height = width * 1.2
+        
+        return CGSize(width: width, height: height)
+    }
+    
+    deinit {
+        cancellables.forEach { $0.cancel() }
+        cancellables.removeAll()
+    }
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        searchTextSubject.send(searchText)
+    }
```

Comment:
Dismiss keyboard after searching finished.
```suggestion
    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
        searchTextSubject.send(searchText)
    }
    
    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
        searchBar.resignFirstResponder()
    }
```